lib/zuora/api.rb

```diff
@@ -118,31 +86,23 @@ def download(export)
     # Upon failure a Zoura::Fault will be raised.
     # @raise [Zuora::Fault]
     def authenticate!
-      response = client.call(:login) do
-        message username: Zuora::Api.instance.config.username,
-                password: Zuora::Api.instance.config.password
-      end
+      response = client.call(:login, message: { username: config.username,  password: config.password })
+
       self.session = Zuora::Session.generate(response.to_hash)
-      client.globals.soap_header({'env:SessionHeader' => {'ins0:Session' => self.session.try(:key) }})
-    rescue Savon::SOAPFault => e
+    rescue Savon::SOAPFault, IOError => e
       raise Zuora::Fault.new(:message => e.message)
     end
 
-    private
-
-    def initialize
-      @config = Config.new
-    end
-
-    def make_client
-      Savon.client(wsdl: fetch_wsdl, soap_version: SOAP_VERSION, log: config.log || true, ssl_verify_mode: :none)
-    end
-
-    def fetch_wsdl
-      config&.wsdl_path || WSDL
+    def client
+      return @client if @client
+
+      @client = Savon.client(
+        wsdl: config&.wsdl_path ? config.wsdl_path : File.expand_path('../../../wsdl/zuora.a.38.0.wsdl', __FILE__),
+        ssl_verify_mode: :none,
```

Comment:
Is setting `ssl_verify_mode: :none` correct? What effect does this have on SSL connections?

Reply:
Upstream sets the same option